src/maker/config.rs

```diff
@@ -61,22 +61,30 @@ impl MakerConfig {
             if path.exists() {
                 parse_toml(path)?
             } else {
+                let default_maker_config_path = get_config_dir().join("maker.toml");
+                if !default_maker_config_path.exists() {
+                    create_default_maker_dirs(&default_maker_config_path);
+                }
                 log::warn!(
                     "Maker config file not found at path : {}, using default config",
                     path.display()
                 );
-                HashMap::new()
+                parse_toml(&default_maker_config_path)?
             }
         } else {
             let default_path = PathBuf::from("maker.toml");
             if default_path.exists() {
                 parse_toml(&default_path)?
             } else {
+                let default_maker_config_path = get_config_dir().join("maker.toml");
+                if !default_maker_config_path.exists() {
+                    create_default_maker_dirs(&default_maker_config_path);
+                }
```

Comment:
This logic needs simplification as follow:

- No custom datadir is provided, so config file should be checked at default datadir, `get_config_dir()`.
- If present, read the config file from there.
- If not, create a default config at that location.

No need to check for `maker.toml` in the current repo directory as that's nonstandard. We should keep the `maker.toml` in repo directory just as an example. But operations should always be done at either custom datadir, or at default datadir.

Reply:
ok ... Got it

Reply:
We are checking this in case the datadir is None.